crates/starknet-server/tests/common/utils.rs

```diff
@@ -72,3 +73,15 @@ pub fn get_events_contract_in_sierra_and_compiled_class_hash()
 
     (sierra_class.flatten().unwrap(), Felt::from(compiled_class_hash).into())
 }
+
+pub fn to_hex_felt<T: LowerHex>(value: &T) -> String {
+    format!("{value:#x}")
+}
+
+pub fn to_num_as_hex<T: LowerHex>(value: &T) -> String {
+    format!("{value:#x}")
```

Comment:
what is the difference between `to_hex_felt` and `to_num_as_hex`

Reply:
The specification distinguishes between [`FELT`](https://github.com/starkware-libs/starknet-specs/blob/v0.4.0/api/starknet_api_openrpc.json#L1266) and [`NUM_AS_HEX`](https://github.com/starkware-libs/starknet-specs/blob/v0.4.0/api/starknet_api_openrpc.json#L1084).

I tried to use `to_num_as_hex` where the specification says to use `NUM_AS_HEX`; the same for `FELT`.

When serializing, they end up looking the same, but in the spirit of logic abstraction, I think it's ok to keep two separate functions, even if right now they have the same implementation.